steamtinkerlaunch

```diff
@@ -17620,6 +17626,16 @@ function dlMod2nexurl {
 			writelog "INFO" "${FUNCNAME[0]} - Starting portable '$RUNPROTON run ${MO}.exe $1'"
 			STEAM_COMPAT_CLIENT_INSTALL_PATH="$STEAM_COMPAT_CLIENT_INSTALL_PATH" STEAM_COMPAT_DATA_PATH="$STEAM_COMPAT_DATA_PATH" "$RUNPROTON" run "${MO}.exe" "$1" 2>&1 | tee /tmp/RUNMO2DL.log
 			cd - >/dev/null || return
+		elif [ -d "${GMO2EXE%/*}" ] && [ -n "$MO2PFX" ] && [ -n "$MO2WINE" ]; then
```

Comment:
Just going to "think out loud" about what this section of the code is doing. Let me know if my understanding is on the right track. For reference, I didn't write *any* of the MO2 code in SteamTinkerLaunch.

To me it seems like this controls how ModOrganizer 2 is started, and it looks to essentially mirror how `startMO2` works? That is, "if instance defined, launch with instance selected, otherwise launch global ModOrganizer 2"

In the case of the URL handler though, it is always using the Game MO2 exe, but if no instance is defined it will just launch without an instance selected. Which probably won't happen when using a URL handler, unless the instance was never set up I guess.

My only "concern" here is in case ModOrganizer 2 could get into a bad startup state where it could never launch, but I don't think that can happen here from reviewing the code :-) This looks pretty safe. GMO2EXE is likely what we want to launch anyway, since that should be what gets started when you launch MO2 in Game Mode to play a game.

Reply:
What it does here is check if the GMO2EXE directory exists, and it has the required MO2PFX and MO2WINE variables, all there of which is coming from the config file it just sourced (the logic is: the steamtinkerlaunch instance that start MO2 dumps its variables into a file, which 'steamtinkerlaunch mo2 u' reads and uses to launch MO2 with the URL specified). So what this does is just check that it has the vars it needs., and if it doesn't it'll fall back to the codepath that's already there (which appears to try a global one, which also fails here, and then it just errors out without this patch).

Reply:
...the trouble with the code that was there is both that it doesn't actually write the file when just starting MO2 from the GUI, but also that the file when written doesn't contain the variables that it needs (namely MO2PFX and MO2WINE). I tried just making sure the file was written initially, but that failed since it didn't have RUNPROTON, which is why I added the new vars and this additional code path which uses them, leaving the old one intact as well since, as you said, it works in some cases - which I'd rather not break (which is also why it does this check *after* checking for the RUNPROTON variant, to make sure the new one can't break the old one).

Reply:
Here's an example of the config file it's exchanging information through, when using the 'start MO2 through the GUI, then use an nxm:// link':
```
[0 zerodogg@browncoats ~]$ cat .config/steamtinkerlaunch/mo2/dldata/newvegas.conf
GMO2EXE="/mnt/ssd4t/SteamLibrary/steamapps/compatdata/22380/pfx/drive_c/Modding/MO2/ModOrganizer.exe"
RUNPROTON=""
MO2PFX="/mnt/ssd4t/SteamLibrary/steamapps/compatdata/22380/pfx"
MO2WINE="/mnt/ssd4t/SteamLibrary/steamapps/common/Proton 9.0 (Beta)/files/bin/wine"
MO2INST=""
STEAM_COMPAT_CLIENT_INSTALL_PATH="/home/zerodogg/.local/share/Steam"
STEAM_COMPAT_DATA_PATH="/mnt/ssd4t/SteamLibrary/steamapps/compatdata/22380"
```

So it's getting all of those vars it's using from there. RUNPROTON isn't used to start mo2, so it's empty, and MO2INST is only used in certain cases, hence the test to see if that too is empty.

Reply:
Makes sense based on my understanding of what the NXM link handlers need and how the existing MO2 logic is in place. Thank you!